vueui/src/components/header/default.vue

```diff
@@ -1,24 +1,60 @@
 <template>
-  <div class="uiTitleWrapper" unselectable="on">
-    <slot/>
+  <div class="uiTitleWrapper" draggable v-on:dragstart="startDragging($event)">
     <div style="clear: both;">
       <div class="uiStatusIcon uiIcon24" :class="statusClass" unselectable="on"/>
-      <div class="uiTitleText" unselectable="on">{{ title }}</div>
-      <div class="uiTitleFluff" unselectable="on"/>
+      <i class="fas ic-bug uiIcon24 uiDebugIcon" :class="debugClass" unselectable="on" @click="activateDebug()"/>
+      <div class="uiTitleText" unselectable="on">{{ d.title }}</div>
+      <div class="uiTitleClose" unselectable="on" @click="closeUI()">×</div>
     </div>
+    <slot/>
   </div>
 </template>
 
 <script>
+import Utils from '../../utils.js';
+import { dragStartHandler } from '../../drag.js';
 export default {
   data () {
-    return this.$root.$data
+    return {
+      debug_flip: 0,
+      d: this.$root.$data,
+    }
   },
   computed: {
     statusClass() {
-      if (this.status == 2) return 'uiStatusGood'
-      if (this.status == 1) return 'uiStatusAverage'
+      if (this.d.status == 2) return 'uiStatusGood'
+      if (this.d.status == 1) return 'uiStatusAverage'
       return 'uiStatusBad'
+    },
+    debugClass() {
+      if (this.d.debug == 1) {
+        if (this.debug_flip == 1) return 'uiDebugOn';
+        if (this.debug_flip == 0) return 'uiDebugOff';
+      }
+      return 'uiNoDebug';
+    },
+    modularComputer() {
+      return this.d.active.includes("mcomputer") ? true : false;
+    }
+  },
+  methods: {
+    closeUI() {
+      Utils.sendToTopic({'vueuiclose': 1});
+    },
+    startDragging($event) {
+      dragStartHandler($event);
+    },
+    activateDebug() {
+      var offset = this.modularComputer ? '72px' : '36px';
```

Comment:
You should be able to easily calculate this dynamically by getting the height of the title wrapper (check `.offsetHeight` or similar on the element). That would allow you to remove the hardcoded check (and simplify lives of people if we ever have more UI styles than simple and modular computer).

Alternatively the calculation could be delegated to the stylesheet by just flipping a class on `body` or the like.

Reply:
Yeah, me at 3 am didn't even think of just using classes for this.